### ISSUE TITLE
Docs vs man

### ISSUE THREAD

**Issue by mauron85** [yannickcr/eslint-plugin-react#893]
How doc options should be understood?

![docsvsman](https://cloud.githubusercontent.com/assets/359450/19183722/f1502a5a-8c79-11e6-8958-0ee0b9084a09.png)

**Comment by ljharb** [comment]
"enabled" is "eslint severity" - which is either "off", "warn", or "error", per eslint docs.

Every eslint rules' first option is configured this way.

**Comment by mauron85** [comment]
Ok, but severity is not mentioned on https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-es6-class.md. It's only in error from eslint atom plugin.  

only options enabled and mode. It is extremely confusing, at least for me.

"prefer-es6-class": [<enabled>, <mode>]

**Comment by ljharb** [comment]
Fair point - want to make a PR to change it to say "severity"?

**Comment by mauron85** [comment]
Ok, but I don't get it. When <enabled> aka severity has options error | warning. What is <mode> for? 
Isn't it same thing?

Possible Severity values: error | warn
Possible mode values: always | never

**Comment by mauron85** [comment]
Ok, I've played with all possible configurations with following outcome (in atom) on es5 react class (React.createClass)

1. severity error, mode: always
```
"react/prefer-es6-class": [2, "always"], 
```
Expected outcome: according severity it should be error but according mode it should warning
Actual: error

**Comment by mauron85** [comment]
My proposal is to remove `<mode>` and leave only severity. Mode "never" IMHO didn't make sense with name of plugin `prefer-es6-class`.

# Enforce ES5 or ES6 class for React Components (prefer-es6-class)

React offers you two way to create traditional components: using the ES5 `React.createClass` method or the new ES6 class system. This rule allow you to enforce one way or another.

## Rule Options

```js
...
"prefer-es6-class": <severity: number>
...
```
### severity off = 0
no error reporting

### severity warning = 1

Will enforce ES6 classes for React Components.

The following patterns are considered warnings:

```js
var Hello = React.createClass({
  render: function() {
    return <div>Hello {this.props.name}</div>;
  }
});
```

The following patterns are not considered warnings:

```js
class Hello extends React.Component {
  render() {
    return <div>Hello {this.props.name}</div>;
  }
}
```

### severity error = 2

Will enforce ES5 classes for React Components

The following patterns are considered errors:

```js
var Hello = React.createClass({
  render: function() {
    return <div>Hello {this.props.name}</div>;
  }
});
```

**Comment by ljharb** [comment]
That's not how an eslint rule can possibly work. `severity` ONLY determines if a report is a warning or an error. Additional options are ALWAYS required to achieve differing behavior.

**Comment by mauron85** [comment]
Before I give up :-)

Did you considered this two scenarios?

1.
<mode> never is like saying prefer ES5 class but name of the plugin is **prefer-es6-class**.

2. "react/prefer-es6-class": [2, "always"], 
according severity it should be error but according mode it should warning

**Comment by ljharb** [comment]
That's a fair point that perhaps the rule could be named better - but I don't think it's particularly important, since most people set it to "always".

The documentation uses the term "warning" because that is how every single eslint rule on the planet refers to them.
